user-interface/src/components/Stopwatch/index.js

```diff
@@ -58,67 +54,56 @@ export default class Stopwatch extends Component {
     });
   }
 
-  handleStart() {
-    if (this.state.startTime === 0) { // Timer is stopped
-      // Change label to pause, and set current countdown to the time now
-      // Also, start the timer using setInterval
-      this.setState({
-        startLabel: 'Pause',
-        startTime: (new Date()).getTime(),
-        intervalId: setInterval(this.formatTime.bind(this), 17),
-      });
+  // Activate/deactivate the timer when start is pressed
+  handleStart = () => {
+    // Timer is stopped
+    if (!this.state.active) {
+      // Set current countdown to the time now and start the timer
+      this.startTime = (new Date()).getTime();
+      this.intervalId = setInterval(this.formatTime, 17);
     } else {
-      // Find out the elapsed time on the timer
-      const elapsed = (new Date()).getTime() - this.state.startTime + this.state.accTime;
-
-      // Stop the timer
-      clearInterval(this.state.intervalId);
-
-      // Change label to start, and reset the start time
-      this.setState({
-        startLabel: 'Start',
-        startTime: 0,
-        accTime: elapsed,
-        intervalId: null,
-      });
+      // Find out the elapsed time on the timer and stop the interval
+      this.accTime = (new Date()).getTime() - this.startTime + this.accTime;
+      clearInterval(this.intervalId);
+      this.intervalId = null;
+      this.startTime = 0;
     }
 
-    this.formatTime();
-  }
+    // Toggle active
+    this.setState({
```

Comment:
Same thing here, consider using a function to update the state instead of an object. Pretty much any time you need to rely on the previous state in order to update new state you should try to use a function!